pkgs/by-name/me/meson/setup-hook.sh

```diff
@@ -1,31 +1,43 @@
+# shellcheck shell=bash disable=SC2206
+
 mesonConfigurePhase() {
     runHook preConfigure
 
+    local flagsArray=()
+
     if [ -z "${dontAddPrefix-}" ]; then
-        mesonFlags="--prefix=$prefix $mesonFlags"
+        flagsArray+=("--prefix=$prefix")
     fi
 
     # See multiple-outputs.sh and meson’s coredata.py
-    mesonFlags="\
-        --libdir=${!outputLib}/lib --libexecdir=${!outputLib}/libexec \
-        --bindir=${!outputBin}/bin --sbindir=${!outputBin}/sbin \
-        --includedir=${!outputInclude}/include \
-        --mandir=${!outputMan}/share/man --infodir=${!outputInfo}/share/info \
-        --localedir=${!outputLib}/share/locale \
-        -Dauto_features=${mesonAutoFeatures:-enabled} \
-        -Dwrap_mode=${mesonWrapMode:-nodownload} \
-        $mesonFlags"
-
-    mesonFlags="${crossMesonFlags+$crossMesonFlags }--buildtype=${mesonBuildType:-plain} $mesonFlags"
```

Comment:
I am not sure if the ``}--`` was any special

Reply:
Bash dark magic. 

The code `${X+Y}` means "if X is not empty then returns Y".
In the case above, it is merely saying "if crossMesonFlags is not empty, put a whitespace after it".